### PR TITLE
improve test structure

### PR DESCRIPTION
**Pull request by macklin-10x**
* Refactors test list declarations to be tuples of an explicit test ID, a comment string, and the args.
* Moves most test list declarations to the single place they are used.
* Simplifies running tests.
* Deletes some unused regression test helper binaries.

### CODE REVIEW

File: enclone_exec/tests/enclone_test3.rs [hunk 1]
```diff
@@ -170,18 +170,23 @@ fn test_site_examples() {
...
-        let out_file = format!("../enclone_help/src/example{}", t + 1);
+    let examples = [
+        (1, r###"BCR=123089 CDR3=CARRYFGVVADAFDIW"###),
+        (
+            2,
+            r###"BCR=123085 GEX=123217 LVARSP=gex,IGHV2-5_g_μ CDR3=CALMGTYCSGDNCYSWFDPW"###,
+        ),
+    ];
+    for (num, args) in examples {
+        let out_file = format!("../enclone_help/src/example{}", num);
```
Comment (sreenathkrishnan) [inline enclone_exec/tests/enclone_test3.rs:181]: We could make the variable inline in format strings
```suggestion
        let out_file = format!("../enclone_help/src/example{num}");
```

File: enclone_testlist/src/lib.rs [hunk 2]
```diff
@@ -3,220 +3,8 @@
...
-}
-pub fn enclone_testdata_public_tcr_human() -> String {
-    include_str!["testdata.public.tcr.human"].to_string()
-}
-pub fn enclone_testdata_public_tcr_mouse() -> String {
-    include_str!["testdata.public.tcr.mouse"].to_string()
-}
-pub fn enclone_testdata_public_gex_human() -> String {
-    include_str!["testdata.public.gex.human"].to_string()
-}
```
Comment (sreenathkrishnan) [inline enclone_testlist/src/lib.rs:20]: These are not used anywhere?

**Review by sreenathkrishnan — APPROVED** [review]
Nice! I like that test functions are self contained with this change. We might want to use insta later since a majority of the tests are really snapshot tests

File: enclone_testlist/src/lib.rs (hunk 2 above)
Comment (macklin-10x) [reply enclone_testlist/src/lib.rs:20]: Indeed, surprisingly unused.

File: enclone_exec/tests/enclone_test3.rs (hunk 1 above)
Comment (macklin-10x) [reply enclone_exec/tests/enclone_test3.rs:181]: Indeed, though this is a lint issue in a lot of places in this codebase, and this was a very mechanical refactor.  I want to get this in without waiting for a full CI cycle.  We can definitely clean this up en masse at some point.

**Comment by macklin-10x** [comment]
Agreed, `insta` would definitely improve this and I'm planning to swap it in as part of a future test refactoring PR.  I'd also really like to refactor the tests so that each individual test case runs as a separate rust test case, allowing the test runner to manage parallelism/reporting/etc.

**Pull request merged**